File: Chapter8/R/Example2.R
# Chapter 8 - Άλλες Σημαντικές Συνεχείς Κατανομές
# Example 2: Εκθετική Κατανομή  X ~ Exp(λ = 0.5)

# Parameter (rate)
lambda <- 0.5

# i. P(X <= 2)
prob1 <- pexp(2, rate = lambda)
cat(sprintf("P(X <= 2) = %.4f\n", prob1))

# ii. P(X > 3)
prob2 <- 1 - pexp(3, rate = lambda)
cat(sprintf("P(X > 3) = %.4f\n", prob2))

# iii. Median
median_x         <- qexp(0.5, rate = lambda)
median_theoretical <- log(2) / lambda
cat(sprintf("Median: %.4f\n", median_x))
cat(sprintf("Theoretical median: %.4f\n", median_theoretical))

# Mean
mean_x <- 1 / lambda
cat(sprintf("Mean: %.4f\n", mean_x))

# Visualization
x <- seq(0, 10, length.out = 1000)
y <- dexp(x, rate = lambda)

plot(x, y, type = 'l', lwd = 2, col = 'blue',
     main = expression(paste('Exponential Distribution, ',
                             lambda, ' = 0.5')),
     xlab = 'x', ylab = 'Density')

# Mark mean and median
abline(v = mean_x,   col = 'red',   lwd = 2, lty = 2)
abline(v = median_x, col = 'green', lwd = 2, lty = 3)

# Shade area for P(X > 3)
x_shade <- x[x >= 3]
y_shade  <- dexp(x_shade, rate = lambda)
polygon(c(3, x_shade, max(x)), c(0, y_shade, 0),
        col = rgb(1, 0, 0, 0.2), border = NA)

legend('topright',
       legend = c('PDF',
                  sprintf('Mean = %.2f',   mean_x),
                  sprintf('Median = %.2f', median_x),
                  sprintf('P(X > 3) = %.2f', prob2)),
       col  = c('blue', 'red', 'green', rgb(1,0,0,0.2)),
       lty  = c(1, 2, 3, NA),
       lwd  = c(2, 2, 2, NA),
       fill = c(NA, NA, NA, rgb(1,0,0,0.2)))

grid()

File: Chapter8/Python/Example2.py
# Chapter 8 - Άλλες Σημαντικές Συνεχείς Κατανομές
# Example 2: Εκθετική Κατανομή  X ~ Exp(λ = 0.5)
#   Note: scipy expon(scale=1/lambda)

import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import expon

# Parameter: expon(scale=1/lambda)
lambda_param = 0.5
dist         = expon(scale=1 / lambda_param)

# i. P(X <= 2)
prob1 = dist.cdf(2)
print(f"P(X <= 2) = {prob1:.4f}")

# ii. P(X > 3)
prob2 = 1 - dist.cdf(3)
print(f"P(X > 3) = {prob2:.4f}")

# iii. Median
median_x          = dist.ppf(0.5)
median_theoretical = np.log(2) / lambda_param
print(f"Median: {median_x:.4f}")
print(f"Theoretical median: {median_theoretical:.4f}")

# Mean
mean_x = dist.mean()
print(f"Mean: {mean_x:.4f}")

# Visualization
x = np.linspace(0, 10, 1000)
y = dist.pdf(x)

plt.figure(figsize=(12, 7))
plt.plot(x, y, 'b-', linewidth=2, label='PDF')

# Mark mean and median
plt.axvline(mean_x,   color='red',   linestyle='--', linewidth=2,
            label=f'Mean = {mean_x:.2f}')
plt.axvline(median_x, color='green', linestyle=':', linewidth=2,
            label=f'Median = {median_x:.2f}')

# Shade area for P(X > 3)
x_shade = x[x >= 3]
y_shade  = dist.pdf(x_shade)
plt.fill_between(x_shade, y_shade, alpha=0.2, color='red',
                 label=f'P(X > 3) = {prob2:.2f}')

plt.xlabel('x', fontsize=12)
plt.ylabel('Density', fontsize=12)
plt.title(r'Exponential Distribution, $\lambda = 0.5$', fontsize=14)
plt.legend(fontsize=10)
plt.grid(alpha=0.3)
plt.tight_layout()
plt.show()
